File: app/src/main/java/pl/nataliana/detectivehardestcase/adapters/DialogsAdapter.java
package pl.nataliana.detectivehardestcase.adapters;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import pl.nataliana.detectivehardestcase.Dialog;
import pl.nataliana.detectivehardestcase.R;

public class DialogsAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    private List<Dialog> dialogs;

    public DialogsAdapter(List<Dialog> dialogs) {
        this.dialogs = dialogs;
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.dialog_list_item, parent, false);
        return new DialogRowHolder(view);
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
        ((DialogRowHolder) holder).setDialog(dialogs.get(position));
    }

    public void addDialog(Dialog dialog) {
        dialogs.add(dialog);
        notifyItemInserted(dialogs.size()-1);
    }

    @Override
    public int getItemCount() {
        return dialogs.size();
    }

    public class DialogRowHolder extends RecyclerView.ViewHolder {

        @BindView(R.id.face_iv)
        ImageView faceImageView;

        @BindView(R.id.name_tv)
        TextView nameLabel;

        @BindView(R.id.dialog_tv)
        TextView dialogLabel;

        public DialogRowHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }

        public void setDialog(Dialog dialog) {
            faceImageView.setImageResource(dialog.getmImageDrawable());
            nameLabel.setText(dialog.getmName());
            dialogLabel.setText(dialog.getmText());
        }
    }
}

File: app/src/main/java/pl/nataliana/detectivehardestcase/adapters/DialogsAdapter.kt
package pl.nataliana.detectivehardestcase.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import butterknife.BindView
import butterknife.ButterKnife
import pl.nataliana.detectivehardestcase.R
import pl.nataliana.detectivehardestcase.model.Dialog

class DialogsAdapter(private val dialogs: MutableList<Dialog>) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.dialog_list_item, parent, false)
        return DialogRowHolder(view)
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        (holder as DialogRowHolder).setDialog(dialogs[position])
    }

    fun addDialog(dialog: Dialog) {
        dialogs.add(dialog)
        notifyItemInserted(dialogs.size - 1)
    }

    override fun getItemCount(): Int {
        return dialogs.size
    }

    inner class DialogRowHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        @BindView(R.id.face_iv)
        internal var faceImageView: ImageView? = null

        @BindView(R.id.name_tv)
        internal var nameLabel: TextView? = null

        @BindView(R.id.dialog_tv)
        internal var dialogLabel: TextView? = null

        init {
            ButterKnife.bind(this, itemView)
        }

        fun setDialog(dialog: Dialog) {
            faceImageView!!.setImageResource(dialog.getmImageDrawable())
            nameLabel!!.text = dialog.getmName()
            dialogLabel!!.text = dialog.getmText()
        }
    }
}
